src/stream/src/executor/source.rs

```diff
@@ -282,9 +319,28 @@ impl<S: StateStore> SourceExecutor<S> {
                     let chunk_with_state =
                         chunk_with_state.map_err(StreamExecutorError::source_error)?;
                     if chunk_with_state.split_offset_mapping.is_some() {
-                        self.state_cache = Some(ConnectorState::from_hashmap(
-                            chunk_with_state.split_offset_mapping.unwrap(),
-                        ));
+                        let mapping: HashMap<String, String> =
+                            chunk_with_state.split_offset_mapping.unwrap();
```

Comment:
Can we initialize `split_offset_mapping` as an `HashMap<SplitId, u64>`? Does it necessarily have to be a String?

Reply:
I think it's necessary. for kafka, the offset is in i64 format but the offset is a string for pulsar and kinesis. Using a string in hashmap to unite them